### ISSUE TITLE
Explain plan for windows shows wrong output types

### ISSUE THREAD

**Issue by dain** [prestodb/presto#5075]
Window functions are showing the wrong output types.  I'm not sure if this is a bigger issue but I think it could result in incorrect results if any code is depending on this.

Here is the new explain plain:
```
presto:tiny> explain (type distributed) SELECT *, 1.0 * sum(x) OVER () FROM (VALUES 1) t(x);
                        Query Plan                        
----------------------------------------------------------
 Fragment 0 [SINGLE]                                      
     Output layout: [x, expr_3]                           
     Output partitioning: SINGLE []                       
     - Output[x, _col1] => [x:bigint, expr_3:double]      
             _col1 := expr_3                              
         - Project => [x:bigint, expr_3:double]           
                 expr_3 := (1.0 * CAST("expr" AS DOUBLE)) 
             - Window[] => [x:bigint, expr:double]     <===== HERE   
                     expr := sum("x")                     
                 - Project => [x:bigint]                  
                         x := CAST("field" AS BIGINT)     
                     - Values => [field:integer]          
                             (1)                          
```

Here is an explain plan from the last release:
```
presto:tiny> explain (type distributed) SELECT *, 1.0 * sum(x) OVER () FROM (VALUES 1) t(x);
                        Query Plan                        
----------------------------------------------------------
 Fragment 0 [SINGLE]                                      
     Output layout: [field, expr_3]                       
     Output partitioning: SINGLE []                       
     - Output[x, _col1] => [field:bigint, expr_3:double]  
             x := field                                   
             _col1 := expr_3                              
         - Project => [field:bigint, expr_3:double]       
                 expr_3 := (1.0 * CAST("expr" AS DOUBLE)) 
             - Window[] => [field:bigint, expr:bigint]     <===== HERE       
                     expr := sum("field")                 
                 - Values => [field:bigint]               
                         (1)                              
```

**Comment by dain** [comment]
@raghavsethi and @martint, I'm going to mark this as a release blocker until we fix it or prove it isn't a problem.

**Comment by dain** [comment]
I found this when rebaseing a PR (https://github.com/prestodb/presto/pull/4699) that depends on this being correct.

**Comment by raghavsethi** [comment]
@martint this is almost certainly a consequence of changing to `analysis.getTypeWithCoercion()`. Let me figure out if reverting that callsite to `analysis.getType()` is the right way to go.

**Comment by raghavsethi** [comment]
Yeah changing it back in window function planning fixes this. Now that I think about it more, I realize what the issue was. Only calls to `analysis.getType()` that stem from requirements to pre-project inputs should use the `getTypeWithCoercion` function.

**Comment by martint** [comment]
Yes, it's due to https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java#L570

It records the type of that allocated symbol as "double" (because that's the type with coercions applied)

**Comment by raghavsethi** [comment]
Should L391 also be changed back? I think that one is also problematic.

**Comment by martint** [comment]
I think that's not needed because that code explicitly deals with the coercions (it unpeels the cast from the expression, then adds one later if necessary)

**Comment by martint** [comment]
Yeah, L391 does the same thing.

**Comment by raghavsethi** [comment]
Cool, so will revert changes to those two lines and put up a diff. IMO the other callsites are all correct but it would be great if you could take a look also.

**Comment by raghavsethi** [comment]
Merged #5077
